changes/tasks.py

```diff
@@ -1468,6 +1469,65 @@ def run(self, vumi_message_id: str, event_type: str, **kwargs) -> None:
 process_whatsapp_system_event = ProcessWhatsAppSystemEvent()
 
 
+class ProcessWhatsAppTimeoutSystemEvent(Task):
+    """
+    Notifies a user that a message we send them on Whatsapp has not been delivered
+    within 7 days, she will be sent an outbound
+    message informing them and give the option to switch to SMS.
+    """
+
+    name = "ndoh_hub.changes.tasks.process_whatsapp_timeout_system_event"
+
+    def handle_undelivered(self, identity_uuid):
+        identity = is_client.get_identity(identity_uuid)
+        # Transform to django language code
+        language = identity["details"]["lang_code"].lower().replace("_", "-")
+        with translation.override(language):
+            text = translation.ugettext(
+                "We see that your MomConnect WhatsApp messages are not being "
+                "delivered. If you would like to receive your messages over "
+                "SMS, reply ‘SMS’."
+            )
+
+        utils.ms_client.create_outbound(
+            {
+                "to_identity": identity_uuid,
+                "content": text,
+                "channel": "JUNE_TEXT",
+                "metadata": {},
+            }
+        )
+
+    def run(
+        self, vumi_message_id: str, timestamp: int, event_type: str, **kwargs
+    ) -> None:
+        try:
+            identity_uuid: str = next(
+                utils.ms_client.get_outbounds({"vumi_message_id": vumi_message_id})[
+                    "results"
+                ]
+            )["to_identity"]
+        except StopIteration:
+            """
+            Outbound with message id doesn't exist, so don't continue
+            """
+            return
+
+        if event_type == "undelivered":
+            d1 = datetime.fromtimestamp(timestamp)
```

Comment:
This timestamp is when the event was created, not when we last sent them the SMS.

The logic should be similar to:
 - We receive this kind of event for the first time, we send them an SMS.
 - If we receive the event a second time, then we need to check when we sent that last SMS. If we sent it less than a month ago, then we do nothing, if it's been more than a month, then we send them that same SMS again.

This means that we need to store when we sent that last SMS somewhere, so that we can look it up when we get another event, and determine if we should send another SMS or not. A good place to store this is probably on the identity, as a field in the identity's `details`

Reply:
makes sense, thank you